lib/routes/1x/index.ts

```diff
@@ -0,0 +1,121 @@
+
+import got from '@/utils/got';
+import { load } from 'cheerio';
+import { art } from '@/utils/render';
+import * as path from 'node:path';
+
+export const handler = async (ctx) => {
+    const { category = 'latest/awarded' } = ctx.req.param();
+    const limit = ctx.req.query('limit') ? Number.parseInt(ctx.req.query('limit'), 10) : 30;
+
+    const rootUrl = 'https://1x.com';
+    const currentUrl = new URL(`gallery/${category}`, rootUrl).href;
+
+    const { data: currentResponse } = await got(currentUrl);
+
+    const $ = load(currentResponse);
+
+    const language = $('html').prop('lang');
+    const apiUrl = new URL(`backend/lm2.php?style=normal&mode=${$('input#lm_mode').prop('value')}`, rootUrl).href;
+
+    const { data: response } = await got(apiUrl);
+
+    const $$ = load(response);
+
+    const items = $$('div.photos-feed-item')
+        .slice(0, limit)
+        .toArray()
+        .map((item) => {
+            item = $(item);
+
+            const title = item.find('span.photos-feed-data-title').first().text() || 'Untitled';
+            const image = item.find('img').prop('src');
+            const author = item.find('span.photos-feed-data-name').first().text();
+
+            const text = `${title} by ${author}`;
+
+            const description = art(path.join(__dirname, 'templates/description.art'), {
+                images: image
+                    ? [
+                          {
+                              src: image,
+                              alt: title,
+                          },
+                      ]
+                    : undefined,
+                description: text,
+            });
+
+            const id = item.find('img[id]').prop('id').split(/-/).pop();
+            const guid = `1x-${id}`;
+
+            return {
+                title,
+                description,
+                link: new URL(`photo/${id}`, rootUrl).href,
+                author,
+                guid,
+                id: guid,
+                content: {
+                    html: description,
+                    text,
+                },
+                image,
+                banner: image,
+                language,
+                enclosure_url: image,
+                enclosure_type: image ? `image/${image.split(/\./).pop()}` : undefined,
+                enclosure_title: title,
+            };
+        });
+
+    const image = new URL($('img.themedlogo').prop('src'), rootUrl).href;
+
+    return {
+        title: $('title').text(),
+        description: $('meta[name="description"]').prop('content'),
+        link: currentUrl,
+        item: items,
+        allowEmpty: true,
+        image,
+        author: $('meta[property="og:site_name"]').prop('content'),
+        language,
+    };
+};
+
+export const route: Route = {
+    path: '/:category{.+}?',
+    name: 'Gallery',
+    url: '1x.com',
+    maintainers: ['nczitzk'],
+    handler,
+    example: '/1x/:category?',
```

Comment:
Please provide actual value instead of route parameters in example.

Reply:
👌